pkg/cmd/adm/register_member.go

```diff
@@ -120,66 +114,210 @@ func registerMemberCluster(ctx *extendedCommandContext, newCommand client.Comman
+	toolchainClusterCR := &toolchainv1alpha1.ToolchainCluster{ObjectMeta: metav1.ObjectMeta{Name: sourceClusterDetails.toolchainClusterName, Namespace: targetClusterDetails.namespace}}
+	_, err = controllerutil.CreateOrUpdate(context.TODO(), targetClusterDetails.client, toolchainClusterCR, func() error {
+
+		// update the tc label
+		labels := toolchainClusterCR.GetLabels()
+		if labels == nil {
+			labels = make(map[string]string)
 		}
-	}()
-	// Create the file
-	file, err := os.CreateTemp("", "add-cluster-*.sh")
+		// TODO drop this "namespace" label as soon as ToolchainCluster controller supports loading data from kubeconfig
+		labels["namespace"] = sourceClusterDetails.namespace
+		toolchainClusterCR.Labels = labels
+		toolchainClusterCR.Spec.APIEndpoint = sourceClusterDetails.apiEndpoint
+		toolchainClusterCR.Spec.SecretRef.Name = secretName
+		if insecureSkipTLSVerify {
+			toolchainClusterCR.Spec.DisabledTLSValidations = []toolchainv1alpha1.TLSValidation{toolchainv1alpha1.TLSAll}
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return err
+	}
+	ctx.Println("Toolchaincluster successfully reconciled")
+	// -- end temporary logic
+
+	return err
+}
+
+func secretName(SANamespacedName runtimeclient.ObjectKey, sourceClusterName string) string {
+	secretName := SANamespacedName.Name + "-" + sourceClusterName
+	return secretName
+}
+
+func newRestClient(kubeConfigPath string) (*rest.RESTClient, error) {
+	restClientConfig, err := clientcmd.BuildConfigFromFlags("", kubeConfigPath)
+	restClientConfig.ContentConfig = rest.ContentConfig{
+		GroupVersion:         &authv1.SchemeGroupVersion,
+		NegotiatedSerializer: scheme.Codecs,
+	}
 	if err != nil {
 		return nil, err
 	}
-	defer func() {
-		if err := file.Close(); err != nil {
-			term.Printlnf(err.Error())
-		}
-	}()
+	restClient, err := rest.RESTClientFor(restClientConfig)
+	if err != nil {
+		return nil, err
+	}
+	return restClient, nil
+}
+
+func generateKubeConfig(token, APIEndpoint, namespace string, insecureSkipTLSVerify bool) *clientcmdapi.Config {
+	// create apiConfig based on the secret content
+	clusters := make(map[string]*clientcmdapi.Cluster, 1)
+	clusters["cluster"] = &clientcmdapi.Cluster{
+		Server:                APIEndpoint,
+		InsecureSkipTLSVerify: insecureSkipTLSVerify,
+	}
+
+	contexts := make(map[string]*clientcmdapi.Context, 1)
+	contexts["ctx"] = &clientcmdapi.Context{
+		Cluster:   "cluster",
+		Namespace: namespace,
+		AuthInfo:  "auth",
+	}
+	authInfos := make(map[string]*clientcmdapi.AuthInfo, 1)
+	authInfos["auth"] = &clientcmdapi.AuthInfo{
+		Token: token,
+	}
 
-	// Write the body to file
-	_, err = io.Copy(file, resp.Body)
-	return file, err
+	clientConfig := &clientcmdapi.Config{
+		Kind:           "Config",
+		APIVersion:     "v1",
+		Clusters:       clusters,
+		Contexts:       contexts,
+		CurrentContext: "ctx",
+		AuthInfos:      authInfos,
+	}
+	return clientConfig
+}
```

Comment:
I guess that you can simplify it to:
```suggestion
func generateKubeConfig(token, apiEndpoint, namespace string, insecureSkipTLSVerify bool) *clientcmdapi.Config {
	return &clientcmdapi.Config{
		Contexts: map[string]*clientcmdapi.Context{
			"ctx": {
				Cluster:   "cluster",
				Namespace: namespace,
				AuthInfo:  "auth",
			},
		},
		CurrentContext: "ctx",
		Clusters: map[string]*clientcmdapi.Cluster{
			"cluster": {
				Server:                apiEndpoint,
				InsecureSkipTLSVerify: insecureSkipTLSVerify,
			},
		},
		AuthInfos: map[string]*clientcmdapi.AuthInfo{
			"auth": {
				Token: token,
			},
		},
	}
}
```

Reply:
yup makes sense thanks!

Reply:
done in https://github.com/kubesaw/ksctl/pull/53/commits/51e2a6ec297ba820306bb184e6384509585e36d3